common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationStringExpressions.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.util;
+
+import com.ibm.icu.text.StringSearch;
+
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * Static entry point for collation aware string expressions.
+ */
+public final class CollationStringExpressions {
+
+  /**
+   * Collation aware string expressions.
+   */
+  public static class Contains {
+    public static boolean containsCollationAware(UTF8String l, UTF8String r, int collationId) {
```

Comment:
here, we could go with `contains` instead of `containsCollationAware`, so that calls look like:
CollationSupport.Contains.contains(l, r, collationId)
CollationSupport.StartsWith.startsWith(l, r, collationId)
CollationSupport.EndsWith.endsWith(l, r, collationId)
etc.

or even something more general such as `execute`, so that we retain some uniformity over different expressions:
CollationSupport.Contains.execute(l, r, collationId)
CollationSupport.StartsWith.execute(l, r, collationId)
CollationSupport.EndsWith.execute(l, r, collationId)
etc.